src/accelerate/utils/dataclasses.py

```diff
@@ -34,6 +34,14 @@
 from .constants import FSDP_AUTO_WRAP_POLICY, FSDP_BACKWARD_PREFETCH, FSDP_STATE_DICT_TYPE
 
 
+@contextmanager
+def clear_os_environ():
+    _old_os_environ = os.environ
+    os.environ = dict()
+    yield
+    os.environ = _old_os_environ
```

Comment:
Let's move this to utils/other so it's by `patch_environment`: https://github.com/huggingface/accelerate/blob/main/src/accelerate/utils/other.py#L117

Reply:
@muellerzr I pushed a new commit:
[move clear_os_environ func to utils/other and rename it](https://github.com/huggingface/accelerate/pull/1738/commits/80c282ffa17e8d759e37862d885b4b7e4761d96f)

In this commit, I find something hard to handle nicely (circular import):
1. I want to use a func `clear_environment` defined in utils.other in utils.dataclasses, utils.dataclasses import utils.other
2. utils.other requires import commands.config.default, thus requires import commands.config
[https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/utils/other.py#L21](https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/utils/other.py#L21)
3. commands.config requires import commands.config.config
[https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/commands/config/__init__.py#L19](https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/commands/config/__init__.py#L19)
4. commands.config.config requires from accelerate.utils import ComputeEnvironment
[https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/commands/config/config.py#L20](https://github.com/huggingface/accelerate/blob/daa1952f4703e1f22cb3788309a5efd88af6f208/src/accelerate/commands/config/config.py#L20)
5. this causes circular import: we are in the process of import utils.dataclasses, but ComputeEnvironment is [visiable](https://github.com/huggingface/accelerate/blob/69e4c3c54da3201eda288b500d138761e7a5221c/src/accelerate/utils/__init__.py#L3) after utils.dataclasses is imported completely.